EditingStyle: prefer enum class for multiple enums

vitorroriz opened the pull request:
#### e5c84e770a0e4eac1774a64b99b34297288bf24a
<pre>
EditingStyle: prefer enum class for multiple enums
<a href="https://rdar.apple.com/157253470">rdar://157253470</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=296774">https://bugs.webkit.org/show_bug.cgi?id=296774</a>

Reviewed by Sammy Gill.

* Source/WebCore/editing/ApplyStyleCommand.cpp:
(WebCore::ApplyStyleCommand::removeInlineStyleFromElement):
(WebCore::ApplyStyleCommand::removeImplicitlyStyledElement):
(WebCore::ApplyStyleCommand::applyInlineStyleToPushDown):
* Source/WebCore/editing/EditingStyle.cpp:
(WebCore::EditingStyle::overrideWithStyle):
(WebCore::EditingStyle::overrideTypingStyleAt):
(WebCore::EditingStyle::triStateOfStyle const):
(WebCore::EditingStyle::conflictsWithImplicitStyleOfElement const):
(WebCore::EditingStyle::extractConflictingImplicitStyleOfAttributes const):
(WebCore::EditingStyle::prepareToApplyAt):
(WebCore::EditingStyle::mergeTypingStyle):
(WebCore::elementMatchesAndPropertyIsNotInInlineStyleDecl):
(WebCore::EditingStyle::wrappingStyleForSerialization):
(WebCore::EditingStyle::mergeStyle):
(WebCore::EditingStyle::legacyFontSize const):
(WebCore::StyleChange::extractTextStyles):
(WebCore::legacyFontSizeFromCSSValue):
* Source/WebCore/editing/EditingStyle.h:
(): Deleted.
* Source/WebCore/editing/InsertTextCommand.cpp:
(WebCore::InsertTextCommand::doApply):
* Source/WebCore/editing/ReplaceSelectionCommand.cpp:
(WebCore::ReplaceSelectionCommand::removeRedundantStylesAndKeepStyleSpanInline):

Canonical link: <a href="https://commits.webkit.org/298122@main">https://commits.webkit.org/298122@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/39c7b1260c46d3f1239eacd589816af232a76dfd

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/114326 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/34071 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/24535 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/120490 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/65040 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/116215 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/34702 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/42632 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/86883 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/41797 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/117274 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/27634 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/102676 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/67275 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/26815 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/20801 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/64180 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/97009 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/20917 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/123700 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/41340 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/30827 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/95705 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/41717 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/98876 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/95489 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24335 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/40635 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/18473 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/37384 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/41220 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/46728 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/40815 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/44121 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/42563 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/77d7266659bf34db35639b3a0b8f99e2e866fbaf. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48801

sammygill commented on Source/WebCore/editing/EditingStyle.h:
```diff
@@ -35,6 +35,7 @@
 #include "CSSPropertyNames.h"
 #include "CSSValueKeywords.h"
 #include "WritingDirection.h"
+#include <cstdint>
```
Is this needed?

sammygill reviewed (approved):
A few of these express the intent of the enum in the type and have redundant information in the values so I wonder if, for example, we should just change the values of `ShouldIgnoreTextOnlyProperties` to `Yes` and `No`

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/ddf9f3d05c2d0ab17e508e14aa455ad1d2661cb0. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48801

vitorroriz committed:
EditingStyle: prefer enum class for multiple enums
rdar://157253470
https://bugs.webkit.org/show_bug.cgi?id=296774

Reviewed by Sammy Gill.

* Source/WebCore/editing/ApplyStyleCommand.cpp:
(WebCore::ApplyStyleCommand::removeInlineStyleFromElement):
(WebCore::ApplyStyleCommand::removeImplicitlyStyledElement):
(WebCore::ApplyStyleCommand::applyInlineStyleToPushDown):
* Source/WebCore/editing/EditingStyle.cpp:
(WebCore::EditingStyle::overrideWithStyle):
(WebCore::EditingStyle::overrideTypingStyleAt):
(WebCore::EditingStyle::triStateOfStyle const):
(WebCore::EditingStyle::conflictsWithImplicitStyleOfElement const):
(WebCore::EditingStyle::extractConflictingImplicitStyleOfAttributes const):
(WebCore::EditingStyle::prepareToApplyAt):
(WebCore::EditingStyle::mergeTypingStyle):
(WebCore::elementMatchesAndPropertyIsNotInInlineStyleDecl):
(WebCore::EditingStyle::wrappingStyleForSerialization):
(WebCore::EditingStyle::mergeStyle):
(WebCore::EditingStyle::legacyFontSize const):
(WebCore::StyleChange::extractTextStyles):
(WebCore::legacyFontSizeFromCSSValue):
* Source/WebCore/editing/EditingStyle.h:
(): Deleted.
* Source/WebCore/editing/InsertTextCommand.cpp:
(WebCore::InsertTextCommand::doApply):
* Source/WebCore/editing/ReplaceSelectionCommand.cpp:
(WebCore::ReplaceSelectionCommand::removeRedundantStylesAndKeepStyleSpanInline):

Canonical link: https://commits.webkit.org/298122@main
